src/net_builtin.c

```diff
@@ -654,16 +665,17 @@ static void read_conn(struct mg_connection *c, struct pkt *pkt) {
     MG_VERBOSE(("%lu SEQ %x -> %x", c->id, mg_htonl(pkt->tcp->seq), s->ack));
     // Advance ACK counter
     s->ack = (uint32_t) (mg_htonl(pkt->tcp->seq) + pkt->pay.len);
-    s->noack += pkt->pay.len;
-    if (s->noack + 1500 >= MIP_TCP_WIN) {
+    s->unacked += pkt->pay.len;
+    // size_t diff = s->acked <= s->ack ? s->ack - s->acked : s->ack;
+    if (s->unacked > MIP_TCP_WIN / 2 && s->acked != s->ack) {
       // Send ACK immediately
-      MG_VERBOSE(("  imm ACK", c->id, mg_htonl(pkt->tcp->seq), s->ack));
+      MG_VERBOSE(("%lu imm ACK %lu", c->id, s->acked));
```

Comment:
`->acked` is the last sent ACK, current data being acked lies within `->acked` and `->ack`. We will actually send `->ack` in our next statement